src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class InsertOptionalDataVisitor : SarifRewritingVisitor
+    {
+        internal IFileSystem s_fileSystem = new FileSystem();
+
+        private Run _run;
+        private bool _overwriteExistingData;
+        private FileRegionsCache _fileRegionsCache;
+        private OptionallyEmittedData _dataToInsert;
+        
+        public InsertOptionalDataVisitor(OptionallyEmittedData dataToInsert, bool overwriteExistingData = true)
```

Comment:
>bool overwriteExistingData = true [](start = 77, length = 33)

You don't need the argument (and I see you don't use it) because `OptionallyEmittedData` has a flag for it.